flepimop/gempyor_pkg/src/gempyor/seir.py

```diff
@@ -16,6 +16,54 @@
 logger = logging.getLogger(__name__)
 
 
+# TO DO: Write documentation for this function
+def check_parameter_positivity(
+    parsed_parameters: np.ndarray,
+    parameter_names: list[str],
+    dates: pd.DatetimeIndex,
+    subpop_names: list[str],
+) -> None:
+    """
+    Identifies and reports earliest negative values for
+    parameters after modifiers have been applied.
+
+    Args:
+        parsed_parameters: An array of parameter values.
+        parameter_names: A list of the names of parameters.
+        dates: A pandas DatetimeIndex containing the dates.
+        subpop_names: A list of the names of subpopulations.
+
+    Raises:
+        ValueError: Negative parameter values were detected.
+
+    Returns:
+        None
+    """
+    if ((parsed_parameters) < 0).any():
+        negative_index_parameters = np.argwhere(parsed_parameters < 0)
```

Comment:
```suggestion
    if len(negative_index_parameters := np.argwhere(parsed_parameters < 0)) > 0:
```
Minor: This use of the walrus allows us to not loop over the array twice.